src/pluginmanager.cpp

```diff
@@ -93,6 +94,8 @@ PluginManager::eventHook(DmtcpEvent_t event, DmtcpEventData_t *data)
 
   case DMTCP_EVENT_VIRTUAL_TO_REAL_PATH:
 
+  case DMTCP_EVENT_PRESUSPEND:
+  case DMTCP_EVENT_PRECHECKPOINT:
```

Comment:
I suggest adding the comment, below.  Events for ckpt/resume/restart are subtle,
since plugins must be called in _reverse_ order during resume/restart.  Let's
document this in the code, so that others will be clearly aware of this issue.
I suggest the following comment:
```
    // The plugins can be thought of as implementing a layered software
    // architecture.  All of the events here occur before writing the checkpoint
    // file.  The plugins are invoked for these events in the natural order.
    // For the resume/restart events below, the plugins are invoked
    // in _reverse_ order.  This is required to support layered software.
    // For an analogous case, see 'man pthread_atfork' with the handlers:
    // (i) prepare, (ii) parent, and (iii) child.
    // Those are analogous to our events for:
    // (i) pre-checkpoint, (ii) resume event, and (iii) restart; respectively.
```